tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,213 @@
+use crate::Args;
+use buildsys_config::DockerArchitecture;
+use clap::Parser;
+use log::{debug, info, trace};
+use pubsys_config::InfraConfig;
+use regex::Regex;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::fs::read_dir;
+use std::path::{Path, PathBuf};
+use tokio::process::Command;
+
+/// Takes a local kit built using buildsys and publishes it
+/// to a vendor specified in Infra.toml
+#[derive(Debug, Parser)]
+pub(crate) struct PublishKitArgs {
+    /// Location of the local kit
+    #[arg(long)]
+    kit_path: PathBuf,
+    /// Vendor to publish kit to
+    #[arg(long)]
+    vendor: String,
+}
+
+struct OciArchive {
+    filename: String,
+    arch: String,
+    version: String,
+}
+
+macro_rules! docker {
+    ($arg: expr) => {{
+        let result = Command::new("docker")
+            .args($arg)
+            .output()
+            .await
+            .context(error::IoSnafu)?;
+        ensure!(
+            result.status.success(),
+            error::PublishFailSnafu {
+                reason: String::from_utf8_lossy(&result.stderr)
+            }
+        );
+        result.stdout
+    }};
+}
+
+pub(crate) async fn run(args: &Args, publish_kit_args: &PublishKitArgs) -> Result<()> {
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let infra_config = InfraConfig::from_path_or_lock(&args.infra_config_path, false)
+        .context(error::ConfigSnafu)?;
+    trace!("Parsed infra config: {:?}", infra_config);
+    ensure!(infra_config.vendor.is_some(), error::NoVendorsSnafu);
+
+    // Fetch the vendor container registry uri
+    let vendor = infra_config
+        .vendor
+        .as_ref()
+        .unwrap()
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    let vendor_registry_uri = vendor.registry.clone();
+    debug!(
+        "Found vendor container registry at uri: {}",
+        vendor_registry_uri
+    );
+
+    // Auto resolve the expected paths for the kit contents archive
+    let kit_path = publish_kit_args.kit_path.as_path();
+    let kit_name = kit_path.file_name().unwrap().to_string_lossy();
+
+    let archives = find_oci_archives(kit_path, &kit_name)?;
+    ensure!(
+        !archives.is_empty(),
+        error::NoArchiveSnafu { path: kit_path }
+    );
+
+    let kit_version = &archives.first().unwrap().version;
+    let mut platform_images = Vec::new();
+    for archive in archives.iter() {
+        let path = kit_path.join(&archive.filename);
+
+        let out = docker!(["load", format!("--input={}", path.display()).as_str(),]);
+        let out = String::from_utf8_lossy(&out);
+        let digest_expression =
+            Regex::new("(?<digest>sha256:[0-9a-f]{64})").context(error::RegexSnafu)?;
+        let caps = digest_expression.captures(&out).unwrap();
```

Comment:
Probably best to capture an error here if Docker does the wrong thing, rather than panic.